tools/healthcheck.py

```diff
@@ -1,130 +1,146 @@
-import unittest
-import time
 import os
-import sys
 import re
+import sys
+import time
+import unittest
+from typing import Any, List, TypedDict
+
+from typing_extensions import NotRequired
+
+from .smoke_tests import SmokeTests
+
+try:
+    from ..evaluation_tests import TestEvaluationFunction  # type: ignore
+except ImportError:
+    TestEvaluationFunction = None
+
+try:
+    from ..preview_tests import TestPreviewFunction  # type: ignore
+except ImportError:
+    TestPreviewFunction = None
+
+
+class JsonTestResult(TypedDict):
+    """JSON-serialisable result of a single unit test."""
+
+    name: str
+    time: NotRequired[int]
 
-from ..tests.requests import TestRequestValidation
-from ..tests.responses import TestResponseValidation
 
-from ..evaluation_tests import TestEvaluationFunction
+JsonTestResults = List[JsonTestResult]
+
+
+class HealthcheckJsonTestResult(TypedDict):
+    """The result object returned by the healthcheck command."""
+
+    tests_passed: bool
+    successes: JsonTestResults
+    failures: JsonTestResults
+    errors: JsonTestResults
 
-"""
-    Extension of the default TestResult class with timing information.
-"""
 
 class HealthcheckResult(unittest.TextTestResult):
-    def __init__(self, *args, **kwargs):
+    """Extension of the default TestResult class with timing information."""
+
+    def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
         self.__path_re = re.compile(r"^[\.\/\w]+\.(\w+\.\w+)$")
 
-        self.__successes_json = []
-        self.__failures_json = []
-        self.__errors_json = []
+        self.__successes_json: JsonTestResults = []
+        self.__failures_json: JsonTestResults = []
+        self.__errors_json: JsonTestResults = []
 
-    def removePathFromId(self, path):
+    def removePathFromId(self, path: str) -> str:
```

Comment:
I think this was because the methods defined in `unittest.TestCase` are all camel case, so it was to be consistent with that. However, I would prefer snake case. I'll updates this.